java-security/src/main/java/com/sap/cloud/security/x509/K8sX509CertExtractor.java

```diff
@@ -0,0 +1,26 @@
+package com.sap.cloud.security.x509;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+
+import static com.sap.cloud.security.x509.X509Constants.FWD_CLIENT_CERT_HEADER;
+
+/**
+ * X509 certificate accessor implementation for Cloud Foundry environment.
+ */
+public class K8sX509CertExtractor implements X509CertExtractor {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(K8sX509CertExtractor.class);
+
+	@Override
+	@Nullable
+	public String getClientCertificate(HttpServletRequest request) {
+		String clientCert = request.getHeader(FWD_CLIENT_CERT_HEADER);
+		LOGGER.debug("{} = {}", FWD_CLIENT_CERT_HEADER, clientCert);
+		return X509Parser.formatBase64Cert(clientCert);
```

Comment:
maybe this can be (is already) handled in the X509Parser.decodeX509?
I guess then the both classes and the interface is not required any longer.